ed25519jni/jvm/src/test/scala/org/zfnd/ed25519/SignatureTest.scala

```diff
@@ -1,22 +1,25 @@
 package org.zfnd.ed25519
 
 import java.security.SecureRandom
-import org.scalatest.{ FlatSpec, MustMatchers }
+import org.scalatest.flatspec.AnyFlatSpec
+import org.scalatest.matchers.must.Matchers
 import scala.collection.mutable.HashSet
 
-class SignatureTest extends FlatSpec with MustMatchers {
+class SignatureTest extends AnyFlatSpec with Matchers {
     private val RANDOM = new SecureRandom()
 
     it must "properly compare Signature objects" in {
         val sig1 = new Array[Byte](Signature.SIGNATURE_LENGTH);
-        do {
-            RANDOM.nextBytes(sig1);
-        } while(!Signature.bytesAreValid(sig1));
+        while ({
+            RANDOM.nextBytes(sig1)
+            !Signature.bytesAreValid(sig1)
+        }) {}
```

Comment:
do/while was removed from Scala 3. I had to change all instances. (The syntax, though.... 🤮 )